homework-2/src/js/controllers/controller.js

```diff
@@ -0,0 +1,47 @@
+import signUp from '../pages/sign-up-page';
+import signIn from '../pages/sign-in-page';
+import restorePassword from '../pages/restore-password-page';
+import patients from '../pages/patients-page';
+
+class Controller {
+  constructor(model, viewsignUp, viewsignIn, viewrestorePassword, viewpatients) {
+    this.model = model;
+    this.views = [viewsignUp, viewsignIn, viewrestorePassword, viewpatients];
+    this.routes = {
+      '/': signUp,
+      '/signIn': signIn,
+      '/restorePassword': restorePassword,
+      '/patients': patients,
+    };
+
+    this.onNavClick('/');
+
+    if (typeof Controller.instance === 'object') {
+      return Controller.instance;
+    }
+    Controller.instance = this;
+    return Controller.instance;
+  }
+
+  onNavClick = (pathname) => {
+    window.history.pushState({}, pathname, window.location.origin + pathname);
+    this.views[0].app.innerHTML = this.routes[pathname];
```

Comment:
As i understood the whole file with <html lang="en">... will be in innerHTML of our index.html?

Reply:
Yeap

Reply:
I cut off part of file 
Now, a container with layout is inserted there